telematic_system/historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/service_manager.py

```diff
@@ -0,0 +1,137 @@
+        log_level (str): Logging level (debug, info, error).
+        log_name (str): Base name for log files.
+        log_path (str): Path to store log files.
+        log_rotation (int): Log file rotation size in bytes.
+        log_handler_type (str): Type of log handler (file, console, all).
+        influx_bucket (str): InfluxDB bucket name.
+        influx_org (str): InfluxDB organization name.
+        influx_token (str): InfluxDB authentication token.
+        influx_url (str): URL of the InfluxDB server.
+        to_str_fields (list): ROS message fields to convert to strings.
+        ignore_fields (list): ROS message fields to ignore.
+        is_nats_connected (bool): Flag indicating NATS connection status.
+        nats_request_topic (str): NATS topic for receiving rosbag processing requests.
+        rosbag_parser (Rosbag2Parser): Object for parsing rosbags.
+    """
+
+    def __init__(self, config=Config()):
+        #NATS client to receive requests from the web ui
+        self.nc = NATS()
+        # List of rosbags to be processed
+        self.rosbag_queue = []
+
+        self.config = config
+
+        # Create rosbag parser object
+        self.rosbag_parser = Rosbag2Parser(config)
+
+        #nats connection status
+        self.is_nats_connected = False
+
+        #TODO: Add Mysql database connection
+        #self.db = _mysql.connect(host="", user="", password="", database="")
+
+
+    async def nats_connect(self):
+        """
+            connect to nats server on EC2
+        """
+        # Nats request callback
+        async def get_file_path_from_nats(msg):
+            self.config.logger.info("Entering process nats request")
+
+            data = msg.data.decode()
+            msg_json_object = json.loads(data)
+
+            rosbag_name = msg_json_object["filename"]
+
+            rosbag_path = Path(self.config.upload_destination_path) / rosbag_name
+            # Add rosbag name to queue
+            self.rosbag_queue.append(rosbag_path)
+
+            # No response sent for the nats request
+
+        async def disconnected_cb():
+            self.config.logger.warn("Got disconnected...")
+
+        async def reconnected_cb():
+            self.config.logger.warn("Got reconnected...")
+
+        async def error_cb(err):
+            self.config.logger.error("{0}".format(err))
+
+        if not self.is_nats_connected:
+            try:
+                await self.nc.connect(self.config.nats_ip_port,
+                                    reconnected_cb=reconnected_cb,
+                                    disconnected_cb=disconnected_cb,
+                                    error_cb=error_cb,
+                                    max_reconnect_attempts=-1)
+                self.config.logger.info("Connected to NATS Server!")
+                self.is_nats_connected = True
+            finally:
+                self.config.logger.info("Client is trying to connect to NATS Server Done.")
```

Comment:
If the code in the `try:` block raises an error, the `finally:` block will erroneously print out that it finished connecting to the server. I suggest we remove the `try: ... finally: ...` block here since we aren't catching errors, and we don't have any "must always run" code.